src/main/java/net/sf/jabref/gui/actions/IntegrityCheckAction.java

```diff
@@ -0,0 +1,85 @@
+package net.sf.jabref.gui.actions;
+
+import java.awt.event.ActionEvent;
+import java.util.List;
+
+import javax.swing.Action;
+import javax.swing.JDialog;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.ListSelectionModel;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.gui.JabRefFrame;
+import net.sf.jabref.logic.integrity.IntegrityCheck;
+import net.sf.jabref.logic.integrity.IntegrityMessage;
+import net.sf.jabref.logic.l10n.Localization;
+
+public class IntegrityCheckAction extends MnemonicAwareAction {
+
+    private static final String ELLIPSES = "...";
+
+
+    private final JabRefFrame frame;
+
+
+    public IntegrityCheckAction(JabRefFrame frame) {
+        this.frame = frame;
+        putValue(Action.NAME, Localization.menuTitle("Check integrity") + ELLIPSES);
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        IntegrityCheck check = new IntegrityCheck(frame.getCurrentBasePanel().getBibDatabaseContext());
+        List<IntegrityMessage> messages = check.checkBibtexDatabase();
+
+        if (messages.isEmpty()) {
+            JOptionPane.showMessageDialog(frame.getCurrentBasePanel(), Localization.lang("No problems found."));
+        } else {
+            // prepare data model
+            Object[][] model = new Object[messages.size()][3];
+            int i = 0;
+            for (IntegrityMessage message : messages) {
+                model[i][0] = message.getEntry().getCiteKey();
```

Comment:
This would require a dedicated table model (I think) and currently it seems like a bit of overkill. But, yes, it would make sense to have some reference to the entry. The method called will not open the entry if there are duplicate keys though.